core/src/main/java/com/microsoft/applicationinsights/TelemetryConfiguration.java

```diff
@@ -220,6 +226,23 @@ public void setRoleName(String roleName) {
         this.roleName = roleName;
     }
 
+    public String getConnectionString() {
+        return connectionString;
+    }
+
+    public void setConnectionString(String connectionString) {
+        try {
+            ConnectionString.parseInto(connectionString, this);
+        } catch (ConnectionStringParseException e) {
+            throw new IllegalArgumentException("Invalid connection string: "+connectionString, e);
```

Comment:
If we assume that a connection string contains Secrets, we shouldn't emit its contents in an Exception.